apps/src/StudioApp.js

```diff
@@ -1549,9 +1571,37 @@ StudioApp.prototype.displayFeedback = function (options) {
     options.feedbackType = this.TestResults.EDIT_BLOCKS;
   }
 
-  this.feedback_.displayFeedback(options, this.requiredBlocks_,
+  if (this.shouldDisplayFeedbackDialog(options)) {
+    // let feedback handle creating the dialog
+    this.feedback_.displayFeedback(options, this.requiredBlocks_,
       this.maxRequiredBlocksToFlag_, this.recommendedBlocks_,
       this.maxRecommendedBlocksToFlag_);
+  } else {
+    // update the block hints lightbulb
+    let missingBlockHints = this.feedback_.getMissingBlockHints(this.recommendedBlocks_);
+    this.displayMissingBlockHints(missingBlockHints);
+
+    // communicate the feedback message to the top instructions via
+    // redux
+    let message = this.feedback_.getFeedbackMessage(options);
+    this.reduxStore.dispatch(setFeedback({ message }));
+  }
+};
+
+/**
+* Whether feedback should be displayed as a modal dialog or integrated
+* into the top instructions
+* @param {{feedbackType: number}} Test results (a constant property of
```

Comment:
actually, this refers to the param options which is an object whose key `feedbackType` should be a number. I copied the jsDoc from `feedback.displayFeedback`; upon closer inspection, it looks like this really should be something like 

```
  * @param {Object} options 
  * @param {number} options.feedbackType
```